torch/csrc/jit/tensorexpr/llvm_codegen.cpp

```diff
@@ -849,8 +855,28 @@ void LLVMCodeGenImpl::visit(const Cast* v) {
   // Scalar casts
   if (srcType->isFPOrFPVectorTy()) {
     if (dstType->isFPOrFPVectorTy()) {
+      // as with eager, convert from Double -> Half by Converting to Float then
+      // Half
+      // TODO: __truncdfhf2
```

Comment:
cc: @penguinwu, you saw this error the other day, this commit fixes it

Reply:
Though my PR triggered the undefined symbol. It also revealed another problem, where `half` was promoted to `double` instead of `float`. I am checking in a fix for the correct type promotion. 

But I have also been wondering why the promotion to double would cause an undefined symbol. It's good to know that that particular problem is fixed too!